internal/validator/main.go

```
@@ -0,0 +1,144 @@
+
+	exchangeClient, err := exchange.NewExchangeClient(network)
+	if err != nil {
+		log.Errorf("cannot create mainnet exchange client: %s", err)
+		os.Exit(1)
+	}
+
+	networkStr = "testnet"
+	network = common.LoadNetwork(networkStr, node)
+	testnetExchangeClient, err := exchange.NewExchangeClient(network)
+	if err != nil {
+		log.Errorf("cannot create testnet exchange client: %s", err)
+		os.Exit(1)
+	}
+
+	finder := MarketFinder{
+		exchangeClient: map[string]exchange.ExchangeClient{
+			"mainnet": exchangeClient,
+			"testnet": testnetExchangeClient,
+		},
+	}
+	for key, m := range metadata {
+		if strings.Contains(strings.ToLower(key), "test") {
+			continue
+		}
+		if m.Denom != "" {
+			for _, market := range m.InjectiveMarkets {
+				if _, ok := TestsMarkets[market]; ok {
+					continue
+				}
+				_, err := finder.findMarketOnAllNetwork(context.TODO(), market)
+				if err != nil {
+					log.Fatalf("%s: market %s not found: %s", key, market, err)
+				} else {
+					log.Infof("%s: market %s found", key, market)
+				}
+			}
+		}
+	}
+
+	log.Infof("token file content is valid")
+}
+
+type MarketFinder struct {
+	exchangeClient map[string]exchange.ExchangeClient
+}
+
+func (f MarketFinder) findMarketOnAllNetwork(ctx context.Context, market string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 1000*time.Second)
+	defer cancel()
+
+	for key, client := range f.exchangeClient {
+		spotMarketInfo := injective_spot_exchange_rpcpb.MarketResponse{}
+		derivativeMarketInfo := injective_derivative_exchange_rpcpb.MarketResponse{}
+		spotMarketInfo, err := client.GetSpotMarket(ctx, market)
+		if err != nil {
+			derivativeMarketInfo, err = client.GetDerivativeMarket(ctx, market)
+			if err != nil &&
+				!strings.Contains(err.Error(), "not found") &&
+				!strings.Contains(err.Error(), "502") {
+				return "", fmt.Errorf("cannot get market %s on %s: %s", market, key, err)
+			}
+			if err != nil {
+				continue
+			}
```

We should never ignore an error or we might have a bad time troubleshooting issues.
I understand that this is just a validation tool and it shouldn't really bother us with bugs but we should always handle the error or log it.

For this case, a simple log would work

---

They are expected too much false positive errors here, but at the end Ill return last error if market is not found